core/domain/DomainFactory.php

```diff
@@ -74,11 +60,43 @@ public static function getShared(FullyQualifiedName $domain_fqcn, array $argumen
      */
     public static function getEventEspressoCoreDomain()
     {
-        $domain = new Domain(
-            new FilePath(EVENT_ESPRESSO_MAIN_FILE),
-            Version::fromString(espresso_version())
-        );
-        LoaderFactory::getLoader()->share('EventEspresso\core\domain\Domain', $domain);
-        return $domain;
+        $fqcn = 'EventEspresso\core\domain\Domain';
+        if(! isset(DomainFactory::$domains[$fqcn])) {
+            DomainFactory::getDomain($fqcn, [EVENT_ESPRESSO_MAIN_FILE, espresso_version()]);
+        }
+        return DomainFactory::$domains[$fqcn];
+    }
+
+
+    private static function getDomain($fqcn, array $arguments)
+    {
+        if(! isset(DomainFactory::$domains[$fqcn])) {
+            if (! isset($arguments[0], $arguments[1])) {
+                throw new InvalidArgumentException(
+                    esc_html__(
+                        'You need to pass at least two arguments, representing the addon plugin file and version, in order to generate a Domain class',
+                        'event_espresso'
+                    )
+                );
+            }
```

Comment:
ya that sounds good... will do thnx

Reply:
this isn't necessary after all...

this method is only called internally (since it's private) and the FQCN has already been validated in the one place it is not hard coded. 

I think the other usage you saw was for `BaristaFactory` which also already validates the FQCN by generating a `FullyQualifiedName` value object that performs multiple checks upon construction.

Reply:
Ok, got it 👍, thanks for double checking.